test/helpers/__init__.py

```diff
@@ -142,7 +149,7 @@ def get_test_gateway(provider, name):
 
     This includes creating a network for the gateway, which is also returned.
     """
-    net_name = 'cb_testgwnet-{0}'.format(get_uuid())
+    net_name = 'cb-testgwnet-{0}'.format(get_uuid())
```

Comment:
@baizhang @chiniforooshan @mbookman Have we got the regex for name validation wrong? I thought gce supported underscores? If not, we may need to update the regex to reflect this.

Reply:
GCE does not allow underscores but from our discussions, iirc, GCE provider was (for the resource name) going to replace the additional symbols allowed within. There's more about this in this comment: https://github.com/gvlproject/cloudbridge/issues/55#issuecomment-322810917

Reply:
GCE allows underscores and dashes in labels, but not in resources (instances, volumes, networks).

I think at the time we discussed https://github.com/gvlproject/cloudbridge/issues/55#issuecomment-322810917, we focused on instance names and agreed to:

- Will not allow upper case letters and will make the restrictions visible to the user.
- Will label field as "host name" or some such to make the restrictions more expected for the user.

The thought was that people generally understand host name restrictions and so pushing this restriction all the way up would be fine.

Would it make sense for each provider to have a validation routine for resource names?

Reply:
This is a good point and it makes a case for not allowing underscores at all.